### PR TITLE
Removed duplicate toleration logic

### LINKED ISSUE #625: ibm-spectrum-scale-csi-attacher-1 pod is in CrashLoopBackOff  status
ibm-spectrum-scale-csi-attacher-1 pod is in CrashLoopBackOff  status

```
[root@api.csi250.cp.fyre.ibm.com ibm-spectrum-scale-container-native]# oc get pods -o wide
NAME                                               READY   STATUS             RESTARTS   AGE    IP              NODE                             NOMINATED NODE   READINESS GATES
csi-scale-fsetdemo-pod-291491-dup                  1/1     Running            0          125m   10.254.22.207   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
csi-scale-fsetdemo-pod-291548                      1/1     Running            0          73m    10.254.22.226   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-2fg7g                       3/3     Running            0          45m    10.17.62.101    worker0.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-8kd8n                       3/3     Running            0          46m    10.17.62.182    worker1.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-attacher-0                  1/1     Running            2          129m   10.254.22.204   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-attacher-1                  0/1     CrashLoopBackOff   49         129m   10.254.8.36     master2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-operator-5dcbb798dd-pqmmj   1/1     Running            0          129m   10.254.22.202   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-provisioner-0               1/1     Running            1          129m   10.254.22.205   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-resizer-0                   1/1     Running            1          129m   10.254.22.203   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-snapshotter-0               1/1     Running            1          129m   10.254.22.206   worker2.csi250.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-xckbn                       3/3     Running            0          45m    10.17.63.101    worker2.csi250.cp.fyre.ibm.com   <none>           <none>

```
```
Events:
  Type     Reason          Age                   From               Message
  ----     ------          ----                  ----               -------
  Normal   Scheduled       40m                   default-scheduler  Successfully assigned ibm-spectrum-scale-csi/ibm-spectrum-scale-csi-attacher-1 to master2.csi250.cp.fyre.ibm.com
  Normal   AddedInterface  40m                   multus             Add eth0 [10.254.8.36/22] from openshift-sdn
  Normal   Created         39m (x5 over 40m)     kubelet            Created container ibm-spectrum-scale-csi-attacher
  Normal   Started         39m (x5 over 40m)     kubelet            Started container ibm-spectrum-scale-csi-attacher
  Warning  Unhealthy       38m (x5 over 40m)     kubelet            Liveness probe failed: Get "http://10.254.8.36:8080/healthz/leader-election": dial tcp 10.254.8.36:8080: connect: connection refused
  Normal   Killing         38m (x5 over 40m)     kubelet            Container ibm-spectrum-scale-csi-attacher failed liveness probe, will be restarted
  Warning  BackOff         5m9s (x145 over 38m)  kubelet            Back-off restarting failed container
  Normal   Pulled          17s (x19 over 40m)    kubelet            Container image "us.gcr.io/k8s-artifacts-prod/sig-storage/csi-attacher:v3.4.0" already present on machine
[root@api.csi250.cp.fyre.ibm.com volsnapshot_testcases]#
```

## Expected behavior
ibm-spectrum-scale-csi-attacher-1 pod should be in Running state

### Data Collection and Debugging
```
[root@api.csi250.cp.fyre.ibm.com volsnapshot_testcases]# oc version
Client Version: 4.8.27
Server Version: 4.8.27
Kubernetes Version: v1.21.6+bb8d50a
[root@api.csi250.cp.fyre.ibm.com volsnapshot_testcases]#
```

```
Images Used
quay.io/ibm-spectrum-scale-dev/ibm-spectrum-scale-csi-driver@sha256:a85cbd5d4ff9f1fe95e90d6ad8a030016bf4907bf1ec3e44a89855cddf0ea933
quay.io/ibm-spectrum-scale-dev/ibm-spectrum-scale-csi-operator@sha256:1a74a2b5f004671c090ab9ea37e0cf69e5b83222c6291f1e30b971fa4bb0f2ea
```

### PR DESCRIPTION
**Pull request by nitishkumar4**
## Pull request checklist

- Fixes #625 

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

- Tolerations are not set correctly in driver and controller pods because of which pods were scheduling only to worker nodes.

## What is the new behavior?

- Removed the duplicate logic of adding tolerations in pod template, duplicate logic was updating tolerations in incorrect format.

## How risky is this change?
- [x] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by nitishkumar4** [comment]
```sh
# oc get pods -o wide -n ibm-spectrum-scale-csi-driver
NAME                                   READY   STATUS    RESTARTS   AGE   IP              NODE                              NOMINATED NODE   READINESS GATES
ibm-spectrum-scale-csi-attacher-0      1/1     Running   2          15m   10.254.4.23     master1.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-attacher-1      1/1     Running   2          15m   10.254.0.42     master0.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-bcvhb           3/3     Running   0          15m   10.17.76.252    worker3.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-bkxsm           3/3     Running   0          15m   10.17.76.63     master0.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-h9wv2           3/3     Running   0          15m   10.17.89.68     worker1.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-kxnpl           3/3     Running   0          15m   10.17.82.1      master1.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-mdrlt           3/3     Running   0          15m   10.17.93.46     worker2.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-provisioner-0   1/1     Running   0          15m   10.254.16.143   worker1.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-resizer-0       1/1     Running   0          15m   10.254.20.23    worker2.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-snapshotter-0   1/1     Running   0          15m   10.254.24.18    worker3.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-wbmv6           3/3     Running   0          15m   10.17.85.69     worker0.csitest.cp.fyre.ibm.com   <none>           <none>
ibm-spectrum-scale-csi-wt2xz           3/3     Running   1          15m   10.17.83.30     master2.csitest.cp.fyre.ibm.com   <none>           <none>
```
**Tolerations in CSIScaleOperator**

```sh
# oc describe cso -n ibm-spectrum-scale-csi-driver
  ...
Tolerations:
    Effect:    NoExecute
    Operator:  Exists
    Effect:    NoSchedule
    Operator:  Exists
    Key:       CriticalAddonsOnly
    Operator:  Exists
...
```

File: operator/controllers/syncer/csi_node.go [hunk 1]
```diff
@@ -118,7 +118,6 @@ func (s *csiNodeSyncer) SyncCSIDaemonsetFn(daemonSetRestartedKey string, daemonS
 	if len(secrets) != 0 {
 		out.Spec.Template.Spec.ImagePullSecrets = secrets
 	}
-	out.Spec.Template.Spec.Tolerations = s.driver.Spec.Tolerations
```
Comment (amdabhad) [inline operator/controllers/syncer/csi_node.go:121]: Do the user specified tolerations from driver CR work when this is removed?

**Review by amdabhad — APPROVED** [review]
LGTM

**Review by Jainbrt — APPROVED** [review]

**Review by deeghuge — APPROVED** [review]

**Review by madhuthorat — APPROVED** [review]
LGTM

**Comment by madhuthorat** [comment]
Closing and re-opening PR to see if Travis can run

**Pull request merged**